test/com/google/javascript/jscomp/PeepholeMinimizeConditionsTest.java

```diff
@@ -337,6 +337,8 @@ public void testNoSwapWithDanglingElse() {
   }
 
   public void testMinimizeHook() {
+    fold("x ? x : y",
+	 "x || y");  
```

Comment:
Can you add test cases to make sure `x() ? x() : y()` and `x.y ? x.y : x.z` are not folded?